app/vmstorage/main.go

```diff
@@ -521,12 +521,22 @@ func writeStorageMetrics(w io.Writer, strg *storage.Storage) {
 	metrics.WriteGaugeUint64(w, `vm_data_size_bytes{type="indexdb/inmemory"}`, idbm.InmemorySizeBytes)
 	metrics.WriteGaugeUint64(w, `vm_data_size_bytes{type="indexdb/file"}`, idbm.FileSizeBytes)
 
+	metrics.WriteCounterUint64(w, `vm_rows_received_by_storage_total`, m.RowsReceivedTotal)
 	metrics.WriteCounterUint64(w, `vm_rows_added_to_storage_total`, m.RowsAddedTotal)
 	metrics.WriteCounterUint64(w, `vm_deduplicated_samples_total{type="merge"}`, m.DedupsDuringMerge)
 	metrics.WriteGaugeUint64(w, `vm_snapshots`, m.SnapshotsCount)
 
+	metrics.WriteCounterUint64(w, `vm_rows_ignored_total{reason="nan_value"}`, m.NaNValueRows)
```

Comment:
I agree that under normal conditions these two are unlikely to happen. And it is hard to come up with the case when a significant number of records is dropped because of this. A bug in the client code that sends NaN values only? A bug in VM that treats all values as NaNs? Both are very unlikely.

Yet the VM code conciders this cases and therefore they should be made visible somehow. Logging may not be enough because not all messages are being logged as I've been told.

Reply:
To put it more simply these would be very useful for debugging.